tests/experimental/evals/functions/test_classify.py

```diff
@@ -34,155 +36,177 @@ def test_llm_classify(monkeypatch: pytest.MonkeyPatch):
+        matcher = M(content__contains=query) & M(content__contains=reference)
+        payload = {
+            "choices": [
+                {
+                    "message": {
+                        "content": response,
+                    },
+                }
+            ],
+        }
+        respx_mock.route(matcher).mock(return_value=httpx.Response(200, json=payload))
 
     with patch.object(OpenAIModel, "_init_tiktoken", return_value=None):
         model = OpenAIModel()
 
-    # without function call in response
-    for message_content in ["relevant", "irrelevant", "\nrelevant ", "unparsable"]:
-        responses.post(
-            "https://api.openai.com/v1/chat/completions",
-            json={"choices": [{"message": {"content": message_content}}]},
-            status=200,
-        )
     result = llm_classify(
         dataframe=dataframe,
         template=RAG_RELEVANCY_PROMPT_TEMPLATE_STR,
         model=model,
         rails=["relevant", "irrelevant"],
-        use_function_calling_if_available=False,
+        verbose=True,
     )
-    labels = ["relevant", "irrelevant", "relevant", NOT_PARSABLE]
-    assert result.iloc[:, 0].tolist() == labels
+
+    expected_labels = ["relevant", "irrelevant", "relevant", NOT_PARSABLE]
+    assert result.iloc[:, 0].tolist() == expected_labels
     assert_frame_equal(
         result,
         pd.DataFrame(
-            index=index,
-            data={"label": ["relevant", "irrelevant", "relevant", NOT_PARSABLE]},
+            data={"label": expected_labels},
         ),
     )
-    del result
-
-    # function call in response
-    for message_content in ["relevant", "irrelevant", "\nrelevant ", "unparsable"]:
-        message = {"function_call": {"arguments": f"{{\n  'response': {message_content}\n}}"}}
-        responses.post(
-            "https://api.openai.com/v1/chat/completions",
-            json={"choices": [{"message": message}]},
-            status=200,
-        )
+
+
+@pytest.mark.respx(base_url="https://api.openai.com/v1/chat/completions")
+def test_llm_classify_with_fn_call(
+    classification_dataframe, monkeypatch: pytest.MonkeyPatch, respx_mock: respx.mock
+):
+    monkeypatch.setenv(OPENAI_API_KEY_ENVVAR_NAME, "sk-0123456789")
+    dataframe = classification_dataframe
+    keys = list(zip(dataframe["query"], dataframe["reference"]))
+    responses = ["relevant", "irrelevant", "\nrelevant ", "unparsable"]
+    response_mapping = {key: response for key, response in zip(keys, responses)}
+
+    for (query, reference), response in response_mapping.items():
+        matcher = M(content__contains=query) & M(content__contains=reference)
+        payload = {
+            "choices": [{"message": {"function_call": {"arguments": {"response": response}}}}]
+        }
+        respx_mock.route(matcher).mock(return_value=httpx.Response(200, json=payload))
+
+    with patch.object(OpenAIModel, "_init_tiktoken", return_value=None):
+        model = OpenAIModel(max_retries=0)
```

Comment:
Curious why need to patch this method here but not elsewhere?

Reply:
I'm not sure I know the other places you are referring to but the unfortunate side-effect is that tiktoken makes network calls so it fails in an isolated CI env. Yet another reason we probably shouldn't have this initialization as part of the construction process, let alone maybe be a part of the model.